Update ImageButton component

azangru opened the pull request:
## Description
The purpose of this PR is to refactor the ImageButton component such that:
- The statusClasses property accepting multiple CSS class names is removed. Instead, the component uses CSS variables for properties that can be styled from the parent
- ImageButton is now using properties of the native button component, thus getting closer to the native button element.

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1618

## Deployment URL(s)
http://image-button-refactoring.review.ensembl.org

ridwanamo commented on src/shared/components/image-button/ImageButton.scss:
```diff
@@ -1,61 +1,61 @@
+  height: var(--image-button-height, auto);
+
+  & > svg {
+    width: var(--image-button-svg-width, 100%);
```
why are we using percentage here instead of an actual value?

ridwanamo replied:
The percentage is relative to the container.

ridwanamo commented on src/shared/components/image-button/ImageButton.tsx:
```diff
@@ -24,52 +28,57 @@ import Tooltip from 'src/shared/components/tooltip/Tooltip';
 export const ImageButton = (props: Props) => {
+  const {
+    status = Status.DEFAULT,
+    description,
```
should description be `description = ''` for the default value because what will happen if no description is passed. will it not be undef instead of empty string?

ridwanamo replied:
Both undef and '' empty string will not show the alt even though accessibility reason there should always be an alt but thats not applicable to us for now.

ridwanamo commented on src/shared/components/image-button/ImageButton.scss:
```diff
@@ -1,61 +1,61 @@
 }
 
 .default {
+  background-color: var(--image-button-default-bg-color, initial);
```
Do we need to set this? if we dont specify this will it not take the browser's default

azangru replied:
Browser is perfectly happy without the fallback. I'm removing it...

![image](https://user-images.githubusercontent.com/6834224/199523635-afae3e73-912d-4402-8045-8e9982b143a9.png)

ridwanamo commented on src/shared/components/image-button/ImageButton.tsx:
```diff
@@ -24,52 +28,57 @@ import Tooltip from 'src/shared/components/tooltip/Tooltip';
-export type Props = {
-  status: ImageButtonStatus;
-  description: string;
+type ChildlessButtonProps = Omit<
```
why do we need this?

ridwanamo replied:
we are not passing children through

azangru replied:
Yes; I also removed the `disabled` property; so that the only way to influence whether the ImageButton is disabled is to pass appropriate props to it.

ridwanamo commented on src/shared/components/image-button/ImageButton.tsx:
```diff
@@ -24,52 +28,57 @@ import Tooltip from 'src/shared/components/tooltip/Tooltip';
+        disabled={status === Status.DISABLED}
+      >
         {typeof props.image === 'string' ? (
           <img src={props.image} alt={props.description} />
```
should this be:
`<img src={Image} alt={description} />`

azangru replied:
`src={Image}`, with capital-case `Image` feels wrong. Let it be as it is.

azangru commented on src/shared/components/image-button/ImageButton.scss:
```diff
@@ -1,61 +1,61 @@
 }
 
 .unselected {
+  background-color: var(--image-button-unselected-bg-color, none);
```
should delete this

ridwanamo commented on stories/shared-components/image-button/ImageButton.stories.tsx:
```diff
@@ -27,97 +26,47 @@ import { Status } from 'src/shared/types/status';
+      </p>
+      <ImageButton
+        status={Status.DEFAULT}
+        description="I am a disabled button; but I will still show a tooltip when moused over"
```
description is wrong here, its not a disabled button.

azangru replied:
✅

ridwanamo commented on src/shared/components/image-button/ImageButton.tsx:
```diff
@@ -24,52 +28,60 @@ import Tooltip from 'src/shared/components/tooltip/Tooltip';
+      >
         {typeof props.image === 'string' ? (
-          <img src={props.image} alt={props.description} />
+          <img src={props.image} alt={description} />
```
There is one more place to replace `props.description` (line 89 inside <Tooltip>)

azangru replied:
✅

azangru commented:
- Moved the parent-customizable `className` property to the top level of the component.
- Removed the `--image-button-width` and `--image-button-height` variables. The parent would need to set the width and height on the ImageButton component directly, by passing it a css class.

imransl commented on src/shared/components/visibility-icon/VisibilityIcon.tsx:
```diff
@@ -39,29 +40,24 @@ type VisibilityIconProps = {
-  let eyeIcon = Eye;
-  let className = styles.default;
+  const { status } = props;
+  const isPartiallySelected = props.status === Status.PARTIALLY_SELECTED;
```
Can't just `status` be used here?

azangru replied:
✅

imransl commented on src/shared/components/visibility-icon/VisibilityIcon.tsx:
```diff
@@ -39,29 +40,24 @@ type VisibilityIconProps = {
-  }
+  const elementClasses = classNames(styles.visibilityIcon, {
+    [styles.visibilityIconPartiallySelected]:
+      status === Status.PARTIALLY_SELECTED
```
`isPartiallySelected` here?

azangru replied:
✅
